### PR TITLE
[charts] Fix themeAugmentation

### PR DESCRIPTION
**Pull request by alexfauquette**
Follow up of #14313

I was looking to add `describeConformance` but those test make assumptions that are not respected by charts. For example it expects `ChartsXAxis` to propagate its props to the root element. Whereas the component take the needed value from its props and render the axis.

In this PR I took care of the general components. Each component has its dedicated commit

### CODE REVIEW

File: packages/x-charts/src/ChartsXAxis/ChartsXAxis.tsx [hunk 1]
```diff
@@ -83,6 +83,12 @@ function addLabelDimension(
   });
 }
 
+const XAxisRoot = styled(AxisRoot, {
+  name: 'MuiChartsXAxis',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+})({});
```
Comment (alexfauquette) [inline packages/x-charts/src/ChartsXAxis/ChartsXAxis.tsx:90]: Looks like a bit of overwork to style a component already styled. But it improve consistency.

I'm considering removing the `AxisRoot` for v8. WHat do you think?

File: packages/x-charts/src/ChartsLegend/ChartsLegend.tsx [hunk 2]
```diff
@@ -59,20 +59,21 @@ const useUtilityClasses = (ownerState: DefaultizedChartsLegendProps & { theme: T
...
-  direction: 'row',
-} as const;
-
 function ChartsLegend(inProps: ChartsLegendProps) {
-  const props: DefaultizedChartsLegendProps = useThemeProps({
-    props: { ...defaultProps, ...inProps },
+  const props = useThemeProps({
+    props: inProps,
     name: 'MuiChartsLegend',
   });
```
Comment (alexfauquette) [inline packages/x-charts/src/ChartsLegend/ChartsLegend.tsx:66]: This `defaultProps` was overriding the one from the themeOverride

File: packages/x-charts/src/ChartsTooltip/ChartsTooltip.tsx [hunk 3]
```diff
@@ -170,7 +171,7 @@ function ChartsTooltip<T extends ChartSeriesType>(props: ChartsTooltipProps<T>)
   return (
     <NoSsr>
       {popperOpen && (
-        <PopperComponent {...popperProps}>
+        <PopperComponent {...popperProps} className={classes.root}>
```
Comment (alexfauquette) [inline packages/x-charts/src/ChartsTooltip/ChartsTooltip.tsx:174]: Before that change, the `PopperComponent` has the styleOverride.root, but the `classes.root` was on the `ChartsTooltipPaper`

With that modification both `styleOverrides.root` and `classes.root` are on the same component

File: packages/x-charts/src/ChartsXAxis/ChartsXAxis.tsx (hunk 1 above)
Comment (JCQuintas) [reply packages/x-charts/src/ChartsXAxis/ChartsXAxis.tsx:90]: What do you mean by removing it? Move the styles to each of the axis?

File: packages/x-charts/src/themeAugmentation/components.d.ts [hunk 4]
```diff
@@ -16,6 +16,11 @@ export interface ChartsComponents<Theme = unknown> {
     styleOverrides?: ComponentsOverrides<Theme>['MuiChartsAxisHighlight'];
   };
   MuiChartsGrid?: {
+    /**
+     * Warning, does not work with LineChart, BarChart and ScatterChart.
+     * Those components skip the grid rendering if it seems unnecessary according to there props.
+     * This `defaultProps` only work if you call `<ChartsGrid />` in a composed chart.
+     */
     defaultProps?: ComponentsProps['MuiChartsGrid'];
```
[the remark this answers is not in the document]
Comment (JCQuintas) [reply packages/x-charts/src/themeAugmentation/components.d.ts:24]: Maybe keep it very simple, like:

```suggestion
    /**
     * The `MuiChartsGrid.defaultProps` only work when using [composition](/x/react-charts/composition/).
     */
     defaultProps?: ComponentsProps['MuiChartsGrid'];
```

An alternative is to have the charts always call `<ChartsGrid />` and chart grid resolves the theme props and decides if it needs to render or not.

[the remark this answers is not in the document]
Comment (alexfauquette) [reply packages/x-charts/src/themeAugmentation/components.d.ts:24]: > An alternative is to have the charts always call <ChartsGrid /> and chart grid resolves the theme props and decides if it needs to render or not.

Rendering the grid is not free. Removing it saves 10ms on rendering. Would need to verify if it's the hooks computation, or the rendering which is responsible

[the remark this answers is not in the document]
Comment (JCQuintas) [reply packages/x-charts/src/themeAugmentation/components.d.ts:24]: Yeah, but we could try to create a proxy renderer eg:

```tsx
function Proxy(inProps) {
  const props = useThemeProps({ props: inProps, name: 'MuiChartsGrid' });
  if (!props.checkExistanceSomehow) return null
  return <ChartsGrid {...props} />
```

[the remark this answers is not in the document]
Comment (alexfauquette) [reply packages/x-charts/src/themeAugmentation/components.d.ts]: I went the other way by creating sub-components for vertical and horizontal. Such that `useTick` is called only if the result will be render.

**Review by JCQuintas — APPROVED** [review]

**Pull request merged**